trace/tests/unit/test__gax.py

```diff
@@ -0,0 +1,377 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import mock
+
+from google.cloud._testing import _GAXBaseAPI
+
+
+class _Base(object):
+    PROJECT = 'PROJECT'
+    PROJECT_PATH = 'projects/%s' % (PROJECT,)
+
+    def _make_one(self, *args, **kw):
+        return self._get_target_class()(*args, **kw)
+
+
+class Test__TraceAPI(_Base, unittest.TestCase):
+
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.trace._gax import _TraceAPI
+
+        return _TraceAPI
+
+    def test_constructor(self):
+        gax_api = _GAXTraceAPI()
+        client = object()
+        api = self._make_one(gax_api, client)
+        self.assertIs(api._gax_api, gax_api)
+        self.assertIs(api.client, client)
+
+    def test_patch_traces(self):
+        from google.cloud.proto.devtools.cloudtrace.v1.trace_pb2 import (
+            TraceSpan, Trace, Traces)
+        from google.cloud._helpers import _datetime_to_pb_timestamp
+
+        from datetime import datetime
+
+        PROJECT = 'PROJECT'
```

Comment:
These are not constants, so please use `lower_snake_case`. I know we're inconsistent about this in this library, but @dhermes and I are actively working to fix that. Please apply throughout.